lib/Diagnostics/Diagnostics.js

```diff
@@ -160,30 +165,34 @@ class Diagnostics extends EventEmitter {
     _updateStreamHealth() {
         return Q.fcall(()=> {
             let streamHealth = StreamHealth.Good;
-            if (this.reportedAlerts.length) {
+            if (this.reportedAlerts.size) {
                 let isPoor = false;
                 let isFair = false;
-                _.forEach(this.reportedAlerts, a => {
-                    if (a.health === StreamHealth.Poor) {
+                this.reportedAlerts.forEach((value) => {
+                    if (value.health === StreamHealth.Poor) {
```

Comment:
why not streamHealth=Math.max(streamHealth,alert.health)?

Reply:
because streamHealth is not an enum, but of type:
`const StreamHealth = {
    Good: 'Good',
    Fair: 'Fair',
    Poor: 'Poor'
};`

Reply:
If I change it to numeric enum the dashboard would have to parse it to string.
I thought we wanted to save this necessity from it